### PR TITLE
don't allow upload/creation of .part files

### PR DESCRIPTION
**Pull request by lordelix**
## Description
don't allow upload/creation of .part files

## Related Issue

https://github.com/owncloud/core/issues/7496

## Motivation and Context
.part is a reserved file-extension in the context of OC

## How Has This Been Tested?

Wrote test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/files/js/files.js [hunk 1]
```diff
@@ -101,7 +101,10 @@
 				throw t('files', '"{name}" is an invalid file name.', {name: name});
 			} else if (trimmedName.length === 0) {
 				throw t('files', 'File name cannot be empty.');
+			} else if (OC.fileIsIgnored(trimmedName)) {
+				throw t('files', '"{name}" is not an allow filetype', {name: name});
```
Comment (PVince81) [inline apps/files/js/files.js]: s/allow/allowed/
s/filetype/file type/

File: core/js/js.js [hunk 2]
```diff
@@ -214,6 +214,22 @@ var OC={
 		return link;
 	},
 
+    /**
+     * Check if a user file is allowed to be handled.
+     * @param {string} file to check
+     */
```
Comment (PVince81) [inline core/js/js.js]: weird indent

**Review by PVince81 — COMMENTED** [review]
This is not enough.
We also need to prevent uploading part files on the Webdav API level.

And the PHP code doesn't have access to that core/js/config.php. So you'll need to add these values elsewhere unfortunately.

File: apps/files/js/files.js (hunk 1 above)
Comment (felixheidecke) [inline apps/files/js/files.js]: Aside from the typos … I'm not a good text-author.
BTW: Is there a text- / wording-guide?

File: core/js/js.js (hunk 2 above)
Comment (felixheidecke) [inline core/js/js.js]: I'll check

File: apps/files/js/files.js (hunk 1 above)
Comment (PVince81) [inline apps/files/js/files.js]: no wording guide so far AFAIK

**Comment by felixheidecke** [comment]
> We also need to prevent uploading part files on the Webdav API level.
Can you assign someone? Whats the procedure here?

**Comment by PVince81** [comment]
@IljaN can you do a separate PR with this single file changes: https://github.com/owncloud/core/pull/7514/files#diff-dd3121bfbeff46c08c30cb0666aa701eR1171

Also add matching integration tests. Thanks

**Comment by PVince81** [comment]
@felixheidecke some JS unit tests failures:
```
Expected null not to equal null.
/var/lib/jenkins/workspace/owncloud-core_core_PR-27149-WPLBBRZ5MQZFQLPMFYZ74GYDKBUZUSYQDL2OK4PRIMDFBQYDKVSQ/apps/files/tests/js/fileUploadSpec.js:86:33
```

**Comment by IljaN** [comment]
@PVince81 I see three changed files in your linked commit?

**Comment by PVince81** [comment]
@IljaN the link https://github.com/owncloud/core/pull/7514/files#diff-dd3121bfbeff46c08c30cb0666aa701eR1171 points at a row of a single file, that's the one to take

**Comment by felixheidecke** [comment]
> And the PHP code doesn't have access to that core/js/config.php. So you'll need to add these values elsewhere unfortunately.

@PVince81 @IljaN I need some advice where to put the config data in order for the backend to get access.

File: core/js/config.php [hunk 3]
```diff
@@ -194,6 +195,11 @@
 			'name' => OC_Util::getTheme()->getName(),
 			'directory' => OC_Util::getTheme()->getDirectory()
 		]
+	),
+	'blacklist_files' => json_encode(
```
Comment (PVince81) [inline core/js/config.php]: Apparently this becomes a global variable "window.blacklist_files", and as we know globals are dirty dirty.

Let's move this into the "oc_config" block which is available from JS through `OC.config.blacklist_files`.

You might also need to add a value for it here for unit testing: https://github.com/owncloud/core/blob/master/core/js/tests/specHelper.js#L95. Note that JS unit tests don't run any PHP code.

@felixheidecke

**Comment by felixheidecke** [comment]
I don't know why the jenkins test doesn't update here. It's not still running ;-) It succeeded :+1: 
@IljaN you can take over now.

File: core/js/js.js [hunk 4]
```diff
@@ -214,6 +214,14 @@ var OC={
 		return link;
 	},
 
+    /**
+     * Check if a user file is allowed to be handled.
+     * @param {string} file to check
```
Comment (PVince81) [inline core/js/js.js]: `@return {boolean} true if allowed, false otherwise

**Comment by PVince81** [comment]
@felixheidecke it just means you're too unlucky 😉 

Sometimes we hit against the Github API rate limit and results cannot be published.

@felixheidecke I'm missing the part in config.php where you read the constant and expose it to the JS side.

**Comment by felixheidecke** [comment]
Hmmm … unlucky me :-P

> @felixheidecke I'm missing the part in config.php where you read the constant and expose it to the JS side.

Do you mean this line? https://github.com/owncloud/core/blob/8a7f7bf209b44c41eecdfcc57bea411d9f5d13c3/core/js/config.php#L158

**Comment by PVince81** [comment]
I scrolled up and down on the full diff looking for that line. But that was before ☕️ 

Saw it now, looks good!

Let's merge this and address the backend part separately.

**Pull request merged**